python/examples/atomistic/3-atomistic-model-with-nl.py

```diff
@@ -0,0 +1,433 @@
+    :align: center
+
+    The simulation engine computes the neighbor lists for the model, which then uses
+    them to predict outputs. This figure is a subset of the figure in
+    :ref:`model-dataflow`.
+
+As example, we will run a 1 ps short molecular dynamics simulation of 125 already
+equilibrated liquid argon atoms interacting via Lennard-Jones within a cutoff of 5 Å.
+The system will be simulated at a temperature of 94.4 K and a mass density of
+1.374 g/cm³. In the end, we will obtain the pair-correlation function :math:`g(r)` of
+the liquid.
+"""
+
+# sphinx_gallery_thumbnail_number = 2
+
+from typing import Dict, List, Optional
+
+# tools for analysis
+import ase.geometry.rdf
+
+# tools to run the simulation and visualization
+import ase.io
+import ase.md
+import ase.neighborlist
+import ase.visualize.plot
+import chemiscope
+import matplotlib.pyplot as plt
+
+# the usual suspects
+import numpy as np
+import torch
+
+from metatensor.torch import Labels, TensorBlock, TensorMap
+from metatensor.torch.atomistic import (
+    MetatensorAtomisticModel,
+    ModelCapabilities,
+    ModelMetadata,
+    ModelOutput,
+    NeighborListOptions,
+    System,
+)
+
+# Integration with ASE calculator for metatensor atomistic models
+from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+
+
+# %%
+#
+# The simulation system
+# ----------------------
+#
+# We load the pre-equilibrated :download:`liquid argon system <liquid-argon.xyz>` from a
+# file.
+
+atoms = ase.io.read("liquid-argon.xyz")
+
+# %%
+#
+# The system was generated based on a 5x5x5 supercell of a simple cubic (sc) cell with a
+# lattice constant of a = 3.641 Å. After initialization of the velocities, the system
+# was run for 100 ps with the same parameters we will use below and the final state can
+# be visualized as
+
+ax = ase.visualize.plot.plot_atoms(atoms, radii=0.5)
+ax.set_xlabel("$\\AA$")
+ax.set_ylabel("$\\AA$")
+plt.show()
+
+# %%
+#
+# The system already has velocities and the expected density of 1.374 g/cm³.
+
+u_to_g = 1.66053906660e-24
+Å_to_cm = 1e-08
+
+mass_density = sum(atoms.get_masses()) / atoms.cell.volume * u_to_g / Å_to_cm**3
+
+print(f"ρ_m = {mass_density:.3f} g/cm³")
+
+# %%
+#
+# Metatensor Neighbor Lists
+# -------------------------
+#
+# Before implementing the actual model, let's take a look at how metatensor stores
+# neighbor lists inside a :py:class:`System` object. We start by computing the neighbor
+# list for our argon systen using ASE.
+
+i, j, S, D = ase.neighborlist.neighbor_list(quantities="ijSD", a=atoms, cutoff=5.0)
+
+# %%
+#
+# The :py:func:`neighbor_list <ase.neighborlist.neighbor_list>` function returns the
+# neighbor indices: quantities ``"i"`` and ``"j"``, the neighbor shifts: ``"S"``, and
+# the distance vectors: ``"D"``. We now stack these together and convert them into the
+# suitable types.
+
+i = torch.from_numpy(i.astype(int))
+j = torch.from_numpy(j.astype(int))
+neighbor_indices = torch.vstack([i, j])
```

Comment:
Can we make this a `n x 2` array instead of `2 x n`? Just to follow metatensor convention from the start